core/src/test/java/com/novoda/noplayer/subtitles/SystemCaptionPreferencesTest.java

```diff
@@ -0,0 +1,289 @@
+package com.novoda.noplayer.subtitles;
+
+import android.content.Context;
+import android.graphics.Color;
+import android.graphics.Typeface;
+import android.os.Build;
+import android.view.accessibility.CaptioningManager;
+import android.view.accessibility.CaptioningManager.CaptionStyle;
+
+import androidx.annotation.NonNull;
+
+import com.google.android.exoplayer2.util.Util;
+import com.novoda.noplayer.test.utils.StaticMutationRule;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import static android.content.Context.CAPTIONING_SERVICE;
+import static android.os.Build.VERSION_CODES.JELLY_BEAN_MR2;
+import static android.os.Build.VERSION_CODES.KITKAT;
+import static android.os.Build.VERSION_CODES.LOLLIPOP;
+import static com.google.android.exoplayer2.text.CaptionStyleCompat.EDGE_TYPE_NONE;
+import static com.google.android.exoplayer2.text.CaptionStyleCompat.EDGE_TYPE_OUTLINE;
+import static com.novoda.noplayer.test.utils.ReflectionFinalMutationUtils.setFinalField;
+import static org.fest.assertions.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(Parameterized.class)
+public class SystemCaptionPreferencesTest {
+
+    private static final int FALLBACK_COLOR = Color.BLUE;
+    private static final float TEXT_SIZE = 20f;
+    private static final int SYSTEM_BACKGROUND_COLOR = Color.RED;
+    private static final int SYSTEM_FOREGROUND_COLOR = Color.GREEN;
+    private static final int SYSTEM_WINDOW_COLOR = Color.GRAY;
+    private static final int SYSTEM_EDGE_TYPE = EDGE_TYPE_OUTLINE;
+    private static final int SYSTEM_EDGE_COLOR = Color.MAGENTA;
+    private static final float SYSTEM_TEXT_SCALE_RATIO = 2f;
+    private static final Typeface SYSTEM_TYPEFACE = mock(Typeface.class);
+    private static final Typeface NO_TYPEFACE = null;
+
+    private final TestCase testCase;
+
+    public SystemCaptionPreferencesTest(TestCase testCase) {
+        this.testCase = testCase;
+    }
+
+    @Parameterized.Parameters(name = "{0}")
+    public static TestCase[] data() {
+        return new TestCase[]{
+            PRE_KITKAT_TEST_CASE,
+            KITKAT_TEST_CASE,
+            POST_LOLLIPOP_TEST_CASE,
+            DISABLED_TEST_CASE,
+            LOCALLY_DISABLED_TEST_CASE
+        };
+    }
+
+    @Rule
+    public final StaticMutationRule mutations = new StaticMutationRule();
```

Comment:
It's good that we have made this a `rule` but I am under the impression that we are missusing it. We only take the advantage of the `after()` routine, but leaving the `before` to be implemented by the user of the rule. 

Can we pass the different mutations we want to perform, as a list of `functions` on the constructor and let them be applied on `before()`?

Reply:
I thought about doing that but we are using Java 1.6 so we can't use lambdas, so this would make it harder/uglier to use :( Maybe if we add Kotlin in the future 😅
The important thing is to clear out the static state so I think this fulfils it's purpose.

One thing I thought of (but didn't want to complicate things) was to have a type that we can pass into the constructor. Happy to have a look at doing something like that:

```Java
new StaticMutationRule(
  mutate(Build.VERSION.class, "SDK_INT", testCase.sdkLevel),
  mutate(Util.class, "SDK_INT", testCase.sdkLevel)
);
```
Where `mutate` is a SFM that creates a class that has an private constructor

Reply:
https://github.com/novoda/no-player/pull/277/commits/3b1719a5b588461d6e26ce212ab39658f5689330